### PR TITLE
Bull Price Refactor

### PR DESCRIPTION
**Pull request by daryakaviani**
Refactor to only calculate squeeth, eth, and crab prices when necessary.

### CODE REVIEW

**Review by haythem96 — APPROVED** [review]
LGTM

File: packages/bull-vault/src/LeverageBull.sol [hunk 1]
```diff
@@ -118,24 +132,24 @@ contract LeverageBull {
...
-            usdcToBorrow =
-                IEulerDToken(dToken).balanceOf(address(this)).wmul(_bullShare).wdiv(ONE.sub(_bullShare)).div(1e12);
+        {
+            if (_bullShare == ONE) {
+                uint256 ethUsdPrice = _getTwap(ethUSDCPool, weth, usdc, TWAP, false);
+                uint256 squeethEthPrice = _getTwap(ethWSqueethPool, wPowerPerp, weth, TWAP, false);
+                uint256 crabUsdPrice = (_ethInCrab.wmul(ethUsdPrice).sub(_squeethInCrab.wmul(squeethEthPrice).wmul(ethUsdPrice)))
+                .wdiv(_totalCrabSupply);
+                uint256 ethToLend = TARGET_CR.wmul(_crabAmount).wmul(crabUsdPrice).wdiv(ethUsdPrice);
+                uint256 usdcToBorrow = ethToLend.wmul(ethUsdPrice).wdiv(TARGET_CR).div(1e12);
```
Comment (aleone) [inline packages/bull-vault/src/LeverageBull.sol:147]: kill ze magic numbers por favor

File: packages/bull-vault/src/LeverageBull.sol [hunk 2]
```diff
@@ -118,24 +132,24 @@ contract LeverageBull {
...
+                return (ethToLend, usdcToBorrow);
+            }
         }
-
-        return (ethToLend, usdcToBorrow);
+        return (IEulerEToken(eToken).balanceOfUnderlying(address(this)).wmul(_bullShare).wdiv(ONE.sub(_bullShare)),
+               IEulerDToken(dToken).balanceOf(address(this)).wmul(_bullShare).wdiv(ONE.sub(_bullShare)).div(1e12));
```
Comment (aleone) [inline packages/bull-vault/src/LeverageBull.sol:152]: no magic pls

**Review by aleone — COMMENTED** [review]
comments same as prior PR, so nothing to do if magic numbers are removed in prior prs

**Pull request merged**